lib/src/csv_parser.cpp

```diff
@@ -103,8 +103,44 @@ Problem CsvParser::load_input() const {
     return problem;
 }
 
-void CsvParser::save_output(const Solution& sln) const {
-    UNUSED(sln)
+void CsvParser::save_output(const std::string& out_path, const Problem& prb,
+    const Solution& sln) const {
+
+	std::ofstream outfile(out_path + std::string("/solution.csv"));
+
+	outfile << "Route;" << "Vehicle;" << "Customer;" << "Arrival;" << "Start;" <<
+		"Finish and leave;" << "Distance from previous;" << "Distance from depot\n";
+
+	char del = ';';
+
+	// for calculating distance from previous customer
+	auto prev_dist = [&prb](const auto& route, size_t i) {
+		return prb.costs[route.second[i].id][route.second[i - 1].id]; 
+	};
+
+	for (size_t i = 0; i < sln.routes.size(); ++i) {
+		const auto& route = sln.routes[i];
+
+		for (size_t j = 1; j < route.second.size() - 1; ++j) { // assuming the first node is depot
+			double prev_cust_dist = prev_dist(route, j);
+			
+			auto veh_id = prb.vehicles[route.first].id;
+			outfile << i << del << veh_id << del
+				<< route.second[j].id << del;
+
+			auto find_veh_id = [&veh_id](const auto& id) { return id.first == veh_id; };
+			// seek for current vehicle in split-delivery
+			auto times = std::find_if(route.second[j].times.begin(),
+				route.second[j].times.end(), find_veh_id);
+
+			outfile << (*times).second;
+
+			outfile << prev_cust_dist
+				<< del << prb.costs[route.second[j].id][0] << "\n";
+		}
+	}
+
+	outfile.close();
```

Comment:
am I wrong or **tabs** still used? (at least, this doesn't look like **4 spaces**)

P.S.: look at lines 143 and 144 for difference